backend/services/implementations/authService.ts

```diff
@@ -163,6 +163,30 @@ class AuthService implements IAuthService {
     }
   }
 
+  async isAuthorizedToChangeRole(
```

Comment:
Could we change this to something more explicit about what it's doing, like `isActiveUserSameAsId` (and invert the logic)?